inference-engine/tests/functional/plugin/shared/src/single_layer_tests/fake_quantize.cpp

```diff
@@ -18,47 +18,133 @@
 
 std::string FakeQuantizeLayerTest::getTestCaseName(testing::TestParamInfo<fqLayerTestParamsSet> obj) {
     fqSpecificParams fqParams;
     InferenceEngine::Precision netPrecision;
     InferenceEngine::SizeVector inputShapes;
     std::string targetDevice;
-    std::tie(fqParams, netPrecision, inputShapes, targetDevice) = obj.param;
+    std::pair<std::string, std::map<std::string, std::string>> config;
+    std::tie(fqParams, netPrecision, inputShapes, targetDevice, config) = obj.param;
     size_t levels;
     std::vector<size_t> constShape;
-    std::tie(levels, constShape) = fqParams;
+    std::vector<float> fqDirectArgs;
+    std::vector<float> inputArg;
+    std::tie(levels, constShape, fqDirectArgs, inputArg) = fqParams;
 
     std::ostringstream result;
     result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
     result << "CS=" << CommonTestUtils::vec2str(constShape) << "_";
     result << "LEVELS=" << levels << "_";
     result << "netPRC=" << netPrecision.name() << "_";
     result << "targetDevice=" << targetDevice;
+    if (!config.first.empty()) {
+        result << "_targetConfig=" << config.first;
+    }
+    if (!fqDirectArgs.empty()) {
+        result << "_fqArgs=" << fqDirectArgs[0] << "_" << fqDirectArgs[1] << "_" << fqDirectArgs[2] << "_" << fqDirectArgs[3];
+    }
+    if (inputArg.size() == 3) {
+        result << "_inputArg=" << inputArg[0] << "_" << inputArg[1] << "_" << inputArg[2];
+    }
     return result.str();
 }
 
 void FakeQuantizeLayerTest::SetUp() {
     fqSpecificParams fqParams;
     std::vector<size_t> inputShape;
+    std::pair<std::string, std::map<std::string, std::string>> config;
     auto netPrecision = InferenceEngine::Precision::UNSPECIFIED;
-    std::tie(fqParams, netPrecision, inputShape, targetDevice) = this->GetParam();
+    std::tie(fqParams, netPrecision, inputShape, targetDevice, config) = this->GetParam();
     InferenceEngine::SizeVector kernel, stride, dilation;
     size_t levels;
     std::vector<size_t> constShape;
-    std::tie(levels, constShape) = fqParams;
+    std::vector<float> fqDirectArg;
+    std::vector<float> inputArg;
+    std::tie(levels, constShape, fqDirectArg, inputArg) = fqParams;
+    if (inputArg.size() == 3) {
+        inputDataMin = inputArg[0];
+        inputDataMax = inputArg[1];
+        inputDataResolution = inputArg[2];
+    }
     auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
     auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
     auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
 
-    auto fq = std::dynamic_pointer_cast<ngraph::opset1::FakeQuantize>(ngraph::builder::makeFakeQuantize(paramOuts[0], ngPrc, levels, constShape));
+    UpdateSeed();
+
+    std::shared_ptr<ngraph::Node> fakeQNode;
+    if (fqDirectArg.empty()) {
+        int32_t ngraphSeed = seed;
+        if (NGRAPH_SEED != USE_CLOCK_TIME) {
```

Comment:
not possible - seed is clock initialized and getting 1  - has 1 chance from 4 billions

Reply:
I would use negative numbers for these flags since generated values will be all positive, so negative one will be never selected. But feel free co close the discussion

Reply:
i had lots of negative seeds see example log:

``` bash
MEM_USAGE=202992KB
Access PluginCache ie core. IE Core use count: 0
Created ie core.
[          ] seed = -1975746804
[          ] ngraphSeed = -1975746804

[          ] seed = -1941737338

[          ] seed = -1939279129

[          ] seed = -1936251160

[          ] seed = -1933212037

```

Reply:
Ok, did not expect for it to be possible, my baD